### PR TITLE
refactor (Android) : Update Spoon core to 5.2.0 & fix plugin for android variant

### PR DESCRIPTION
**Pull request by paillardf**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,20 +3,36 @@ apply plugin: 'maven'
 
 repositories {
     mavenCentral()
-    maven {
-        url 'https://repo.eclipse.org/content/repositories/eclipse-staging/'
+    jcenter()
```
Comment (GerardPaligot) [inline build.gradle]: Necessary?

File: src/main/groovy/fr/inria/gforge/spoon/SpoonAndroidPlugin.groovy [hunk 2]
```diff
@@ -25,41 +27,60 @@ class SpoonAndroidPlugin implements Plugin<Project> {
...
+                            }
+
                 }
                 // Inserts spoon task before compiling.
                 compileJavaTask.dependsOn spoonTask
             }
-        })
+        }
+
+        )
```
Comment (GerardPaligot) [inline src/main/groovy/fr/inria/gforge/spoon/SpoonAndroidPlugin.groovy:82]: Bad indentation somewhere.

File: src/main/groovy/fr/inria/gforge/spoon/SpoonExtension.groovy [hunk 3]
```diff
@@ -13,7 +13,7 @@ class SpoonExtension {
     def File outFolder
 
     /** Tells to spoon that it must preserve formatting of original source code. */
-    def boolean preserveFormatting
+    def boolean preserveFormatting = true;
```
Comment (GerardPaligot) [inline src/main/groovy/fr/inria/gforge/spoon/SpoonExtension.groovy]: This shouldn't be its default value.

File: src/test/resources/android-app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (GerardPaligot) [inline src/test/resources/android-app/src/main/AndroidManifest.xml:1]: Necessary?

Comment (paillardf) [inline src/test/resources/android-app/src/main/AndroidManifest.xml:1]: New Android plugin need a AndroidManifest file to be applied on a projet. If you have any better idea I am interested to hear it

**Pull request merged**